token/program-2022/src/processor.rs

```diff
@@ -371,6 +372,12 @@ impl Processor {
             return Err(TokenError::MintMismatch.into());
         }
 
+        if let Ok(extension) = dest_account.get_extension_mut::<MemoTransfer>() {
+            if extension.require_incoming_transfer_memos != 0 {
+                // TODO: use get_processed_instructions syscall to check for memo
```

Comment:
Just noting this is where the update will be needed when https://github.com/solana-labs/solana/pull/22859 is available. Until then, all transactions into accounts requiring transfer-memos will succeed.